src/Microsoft.CodeQuality.Analyzers/Core/ApiDesignGuidelines/StringBuilderAppendShouldNotTakeSubstring.cs

```diff
@@ -0,0 +1,135 @@
+        private static readonly LocalizableString s_localizableMessageDefault2 = new LocalizableResourceString(
+            nameof(MicrosoftApiDesignGuidelinesAnalyzersResources.StringBuilderShouldUseSubstringOverloadWithTwoParameterDefault),
+            MicrosoftApiDesignGuidelinesAnalyzersResources.ResourceManager,
+            typeof(MicrosoftApiDesignGuidelinesAnalyzersResources));
+        private static readonly LocalizableString s_localizableDescription2 = new LocalizableResourceString(
+            nameof(MicrosoftApiDesignGuidelinesAnalyzersResources.StringBuilderShouldUseSubstringOverloadWithTwoParameterDescription),
+            MicrosoftApiDesignGuidelinesAnalyzersResources.ResourceManager, 
+            typeof(MicrosoftApiDesignGuidelinesAnalyzersResources));
+
+        internal static DiagnosticDescriptor RuleReplaceOneParameter = new DiagnosticDescriptor(
+            RuleIdOneParameterId,
+            s_localizableTitle,
+            s_localizableMessageDefault,
+            DiagnosticCategory.Usage,
+            DiagnosticHelpers.DefaultDiagnosticSeverity,
+            isEnabledByDefault: DiagnosticHelpers.EnabledByDefaultIfNotBuildingVSIX,
+            description: s_localizableDescription,
+            helpLinkUri: "", // TODO (might require an addition to docs repository?
+            customTags: WellKnownDiagnosticTags.AnalyzerException);
+
+        internal static DiagnosticDescriptor RuleReplaceTwoParameter = new DiagnosticDescriptor(
+            RuleIdTwoParameterId,
+            s_localizableTitle2,
+            s_localizableMessageDefault2,
+            DiagnosticCategory.Usage,
+            DiagnosticHelpers.DefaultDiagnosticSeverity,
+            isEnabledByDefault: DiagnosticHelpers.EnabledByDefaultIfNotBuildingVSIX,
+            description: s_localizableDescription2,
+            helpLinkUri: "", // TODO (might require an addition to docs repository?
+            customTags: WellKnownDiagnosticTags.AnalyzerException);
+
+        public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics => ImmutableArray.Create(RuleReplaceOneParameter, RuleReplaceTwoParameter);
+
+        public override void Initialize(AnalysisContext context)
+        {
+            context.EnableConcurrentExecution();
+            context.ConfigureGeneratedCodeAnalysis(GeneratedCodeAnalysisFlags.None);
+
+            context.RegisterSyntaxNodeAction(this.Action, SyntaxKind.InvocationExpression);
+        }
+
+        private void Action(SyntaxNodeAnalysisContext context)
+        {
+            if (context.Node is InvocationExpressionSyntax invocation)
+            {
+                var invokedSymbol = context.SemanticModel.GetSymbolInfo(invocation);
+                var invokedClass = invokedSymbol.Symbol.ContainingSymbol;
+                var invokedMethod = invokedSymbol.Symbol.Name;
+                var namedType = invokedClass as INamedTypeSymbol;
+
+                if (namedType.ContainingNamespace.Name == "Text" 
```

Comment:
This will NRE if `invokedClass` is not `INamedTypeSymbol`. If it will always be that type, use a direct cast rather than `as`. Consider using pattern matching.

Reply:
right, that should be included as first operand in the if condition below. Thanks, fixing it.